dvc/external_repo.py

```diff
@@ -62,16 +62,33 @@ def _external_repo(url=None, rev=None, cache_dir=None):
 
     # Adjust new clone/copy to fit rev and cache_dir
     repo = Repo(new_path)
+    original_repo = None
     try:
         if rev is not None:
             repo.scm.checkout(rev)
+        # check if the URL is local and no default remote
+        # add default remote pointing to the original repo's cache location
+        if not os.path.isdir(url):
+            # Adjust original repo for pointing remote towards its' cache
+            original_repo = Repo(url)
+            rconfig = RemoteConfig(original_repo.config)
```

Comment:
No need to read original repo directly, we have our clone already (`repo`) and we can use it to check if default remote is set and can then set default remote in it, we def shouldn't modify the original repo directly.